### PR TITLE
fix: can't change group role on api v7 - WPB-15054

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-15054" title="WPB-15054" target="_blank">BugWPB-15054</a>  [iOS] Use correct v7 endpoint for making someone group admin
  </td></table>
  <br />
 

### Issue

When using API V7, if I try to change the role of a group participant (e.g make an admin), then it fails. The cause is because the endpoint changed slightly in V7. The fix is to update the endpoint path.

### Testing

1. Log in, select API V7, create a group with one other member.
2. Make the user an admin
3. Assert the other user is an admin.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/UpdateRoleActionHandler.swift [hunk 1]
```diff
@@ -36,7 +36,23 @@ class UpdateRoleActionHandler: ActionHandler<UpdateRoleAction> {
             return nil
         }
 
-        let path = "/conversations/\(conversationId.transportString())/members/\(userId.transportString())"
+        let path: String
+
+        switch apiVersion {
+        case .v0, .v1, .v2, .v3, .v4, .v5, .v6:
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/UpdateRoleActionHandler.swift:42]: do we need a change in WireAPI too?

**Review by caldrian — APPROVED** [review]

**Review by netbe — APPROVED** [review]
👍, but left a comment do we need to update for WireAPI? - redirect to 3.115.0 release branch

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/UpdateRoleActionHandler.swift (hunk 1 above)
Comment (johnxnguyen) [reply wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/UpdateRoleActionHandler.swift:42]: no, the endpoint isn't implemented yet

**Comment by samwyndham** [comment]
Removing myself as reviewer as this PR is already approved

**Pull request merged**